lib/pbench/client/oidc_admin.py

```diff
@@ -0,0 +1,148 @@
+            last_name: Optional first name of the user,
+
+        Returns:
+            Response from the request.
+        """
+        admin_token = self.get_admin_token().get("access_token")
+        url_path = f"/admin/realms/{self.OIDC_REALM}/users"
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {admin_token}",
+        }
+        data = {
+            "username": username,
+            "email": email,
+            "emailVerified": True,
+            "enabled": True,
+            "firstName": first_name,
+            "lastName": last_name,
+            "credentials": [
+                {"type": "password", "value": password, "temporary": False}
+            ],
+        }
+        response = self.post(path=url_path, data=json.dumps(data), headers=headers)
+        return response
+
+    def user_login(self, client_id: str, username: str, password: str) -> dict:
+        """pbench-server realm user login on a specified client.
+
+        Args:
+            client_id: client_name to use in the request
+            username: username of the user logging in
+            password: OIDC password
+
+        Returns:
+            access_token json payload
+
+            { 'access_token': <access_token>,
+              'expires_in': 60, 'refresh_expires_in': 1800,
+              'refresh_token': <refresh_token>,
+              'token_type': 'Bearer',
+              'not-before-policy': 0,
+              'session_state': '8f558797-50e7-496d-bb45-3b5ac9fdcddb',
+              'scope': 'profile email'}
+
+        """
+        url_path = f"/realms/{self.OIDC_REALM}/protocol/openid-connect/token"
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        data = {
+            "client_id": client_id,
+            "grant_type": "password",
+            "scope": "profile email",
+            "username": username,
+            "password": password,
+        }
+        response = self.post(path=url_path, data=data, headers=headers)
+        return response.json()
+
+    def get_user(self, username: str, token: str) -> dict:
+        """Get the OIDC user representation dict.
+
+        Args:
+            username: username to query
+            token: access_token string to validate
+
+        Returns:
+            User dict representation
+
+            {'id': '37117992-a3de-43f7-b844-e6ee178e9965',
+            'createdTimestamp': 1675981768951,
+            'username': 'admin',
+            'enabled': True,
+            'totp': False,
+            'emailVerified': False,
+            'disableableCredentialTypes': [],
+            'requiredActions': [],
+            'notBefore': 0,
+            'access': {'manageGroupMembership': True, 'view': True, 'mapRoles': True, 'impersonate': True, 'manage': True}
+            ...
+            }
+        """
+        response = self.get(
+            f"admin/realms/{self.OIDC_REALM}/users?username={username}",
```

Comment:
In general I'd prefer to use the query parameter dict rather than hardcoding the query syntax, which is in general harder to maintain over time... but I don't remember (and at the moment I'm too lazy to check) whether your client `get` wrapper supports that?